### ISSUE TITLE
`e: Compilation failed: null` on 1.3.50 on native(iOSx64 and iOSArm64) only

### ISSUE THREAD

**Issue by sujitpoudel** [Kotlin/kotlinx.serialization#581]
**Describe the bug**
I have this super helpful `e: Compilation failed: null` error message while compiling a Serializable class with a super type, only on native (iOS). The same logic, and code works without any issue on android project. But when putting this in multiplatform project it fails in native compilation with that error message.

**To Reproduce**
I've a very simple serializable class like:
```
@Serializable
class AccessPolicy : Model<AccessPolicy>(serializer())
```

The Model class looks like:
```
@Serializable
abstract class Model<T : Model<T>>(var serializer: KSerializer<T>?) {
    @Suppress("unused")
    constructor() : this(null)
}
```

Compiling it, I get the following stacktrace:
```
 e: Compilation failed: null

 * Source files: ClientResponse.kt, <A whole bunch of other files>
 * Compiler version info: Konan: 1.3.50 / Kotlin: 1.3.50
 * Output kind: LIBRARY

e: kotlin.KotlinNullPointerException
	at org.jetbrains.kotlin.backend.common.serialization.IrModuleSerializer.serializeTypeArguments(IrModuleSerializer.kt:169)
	at org.jetbrains.kotlin.backend.common.serialization.IrModuleSerializer.serializeMemberAccessCommon(IrModuleSerializer.kt:365)
	at org.jetbrains.kotlin.backend.common.serialization.IrModuleSerializer.serializeDelegatingConstructorCall(IrModuleSerializer.kt:467)
	at org.jetbrains.kotlin.backend.common.serialization.IrModuleSerializer.serializeExpression(IrModuleSerializer.kt:794)
	at org.jetbrains.kotlin.backend.common.serialization.IrModuleSerializer.serializeStatement(IrModuleSerializer.kt:848)
	at org.jetbrains.kotlin.backend.common.serialization.IrModuleSerializer.serializeBlockBody(IrModuleSerializer.kt:306)
	at org.jetbrains.kotlin.backend.common.serialization.IrModuleSerializer.serializeStatement(IrModuleSerializer.kt:851)
	at org.jetbrains.kotlin.backend.common.serialization.IrModuleSerializer.serializeIrFunctionBase(IrModuleSerializer.kt:929)
	at org.jetbrains.kotlin.backend.common.serialization.IrModuleSerializer.serializeIrConstructor(IrModuleSerializer.kt:943)
	at org.jetbrains.kotlin.backend.common.serialization.IrModuleSerializer.serializeDeclaration(IrModuleSerializer.kt:1100)
	at org.jetbrains.kotlin.backend.common.serialization.IrModuleSerializer.serializeIrDeclarationContainer(IrModuleSerializer.kt:1041)
	at org.jetbrains.kotlin.backend.common.serialization.IrModuleSerializer.serializeIrClass(IrModuleSerializer.kt:1068)
	at org.jetbrains.kotlin.backend.common.serialization.IrModuleSerializer.serializeDeclaration(IrModuleSerializer.kt:1112)
	at org.jetbrains.kotlin.backend.common.serialization.IrModuleSerializer.serializeIrFile(IrModuleSerializer.kt:1148)
	at org.jetbrains.kotlin.backend.common.serialization.IrModuleSerializer.serializeModule(IrModuleSerializer.kt:1201)
	at org.jetbrains.kotlin.backend.common.serialization.IrModuleSerializer.serializedIrModule(IrModuleSerializer.kt:1220)
	at org.jetbrains.kotlin.backend.konan.ToplevelPhasesKt$serializerPhase$1.invoke(ToplevelPhases.kt:210)
	at org.jetbrains.kotlin.backend.konan.ToplevelPhasesKt$serializerPhase$1.invoke(ToplevelPhases.kt)
	at org.jetbrains.kotlin.backend.common.phaser.PhaseBuildersKt$namedOpUnitPhase$1.invoke(PhaseBuilders.kt:115)
	at org.jetbrains.kotlin.backend.common.phaser.PhaseBuildersKt$namedOpUnitPhase$1.invoke(PhaseBuilders.kt:113)
	at org.jetbrains.kotlin.backend.common.phaser.AbstractNamedPhaseWrapper$runBody$1.invoke(CompilerPhase.kt:128)
	at org.jetbrains.kotlin.backend.common.phaser.CompilerPhaseKt.downlevel(CompilerPhase.kt:24)
	at org.jetbrains.kotlin.backend.common.phaser.AbstractNamedPhaseWrapper.runBody(CompilerPhase.kt:127)
	at org.jetbrains.kotlin.backend.common.phaser.AbstractNamedPhaseWrapper.invoke(CompilerPhase.kt:105)
	at org.jetbrains.kotlin.backend.common.phaser.CompositePhase.invoke(PhaseBuilders.kt:28)
	at org.jetbrains.kotlin.backend.common.phaser.AbstractNamedPhaseWrapper$runBody$1.invoke(CompilerPhase.kt:128)
	at org.jetbrains.kotlin.backend.common.phaser.CompilerPhaseKt.downlevel(CompilerPhase.kt:24)
	at org.jetbrains.kotlin.backend.common.phaser.AbstractNamedPhaseWrapper.runBody(CompilerPhase.kt:127)
	at org.jetbrains.kotlin.backend.common.phaser.AbstractNamedPhaseWrapper.invoke(CompilerPhase.kt:105)
	at org.jetbrains.kotlin.backend.common.phaser.CompilerPhaseKt.invokeToplevel(CompilerPhase.kt:42)
	at org.jetbrains.kotlin.backend.konan.KonanDriverKt.runTopLevelPhases(KonanDriver.kt:27)
	at org.jetbrains.kotlin.cli.bc.K2Native.doExecute(K2Native.kt:79)
	at org.jetbrains.kotlin.cli.bc.K2Native.doExecute(K2Native.kt:34)
	at org.jetbrains.kotlin.cli.common.CLICompiler.execImpl(CLICompiler.kt:84)
	at org.jetbrains.kotlin.cli.common.CLICompiler.execImpl(CLICompiler.kt:42)
	at org.jetbrains.kotlin.cli.common.CLITool.exec(CLITool.kt:104)
	at org.jetbrains.kotlin.cli.common.CLITool.exec(CLITool.kt:82)
	at org.jetbrains.kotlin.cli.common.CLITool.exec(CLITool.kt:50)
	at org.jetbrains.kotlin.cli.common.CLITool$Companion.doMainNoExit(CLITool.kt:215)
	at org.jetbrains.kotlin.cli.common.CLITool$Companion.doMain(CLITool.kt:207)
	at org.jetbrains.kotlin.cli.bc.K2Native$Companion$main$1.invoke(K2Native.kt:223)
	at org.jetbrains.kotlin.cli.bc.K2Native$Companion$main$1.invoke(K2Native.kt:220)
	at org.jetbrains.kotlin.konan.util.UtilKt.profileIf(Util.kt:26)
	at org.jetbrains.kotlin.konan.util.UtilKt.profile(Util.kt:21)
	at org.jetbrains.kotlin.cli.bc.K2Native$Companion.main(K2Native.kt:222)
	at org.jetbrains.kotlin.cli.bc.K2NativeKt.main(K2Native.kt:304)
	at org.jetbrains.kotlin.cli.utilities.MainKt.main(main.kt:16)


> Task :compileKotlinIos64 FAILED

e: Compilation failed: null

Execution failed for task ':compileKotlinIos64'.
> Process 'command '/Library/Java/JavaVirtualMachines/jdk1.8.0_171.jdk/Contents/Home/bin/java'' finished with non-zero exit value 2

```
**Expected behavior**
No compilation error, and should work like it works on android platform compilation.

**Environment**
 - Kotlin version: 1.3.50
 - Library version: 0.13.0
 - Kotlin platforms: android, Native(iOSx64, iOSArm64)
 - Gradle version: 5.6.2

**Comment by pdvrieze** [comment]
Is `AccessPolicy` serializable? For things to work it has to be serializable.

**Comment by sandwwraith** [comment]
This looks like #525 to me. Will be fixed in Kotlin 1.3.60
